test/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 itk_module_test()
 
 set(IOOMEZarrNGFFTests
+  itkOMEZarrNGFFHTTPTest.cxx
```

Comment:
Could you add test pointing to the one of the (smallest) public data, from e.g. https://www.openmicroscopy.org/2020/11/04/zarr-data.html ?

Reply:
Definitely, I wasn't sure where to find public test data on S3 before. Will take a look.

Reply:
I will add that there could be complications in the future from referencing test data that we do not control, such as if the data is moved or removed.

I did look briefly at fetching from `data.kitware.com`. But, files and folders are referenced by hash in URLs and not by directory structure, so the `data.kitware.com` URL interface is not compatible with tensorstore `http` retrieval.

Reply:
Using a server which we don't control is non-ideal, but it is better than not having a test at all. And even if data is moved, an update to the new location should not be hard.

Reply:
OME uses zarr v2, and it does not seem likely it will switch to v3 soon.

You should ask about float issue here:
https://github.com/google/tensorstore

Reply:
I would recommend this dataset:

https://uk1s3.embassy.ebi.ac.uk/idr/zarr/v0.4/idr0062A/6001240.zarr

which is a commonly used test dataset, also v0.4.

fill_value should likely be a number instead of a string, but we should ensure that Tensorstore can handle a string.

CC @jbms @joshmoore @jakirkham

Reply:
> OME uses zarr v2, and it does not seem likely it will switch to v3 soon.

Sounds good. You are correct, I mistakenly took the URL to indicate the Zarr format, but based on Matt's link it looks like "v0.1" is the OME-Zarr version. The Zarr header confirms that data is indeed Zarr v2 format.

> You should ask about float issue here:

Will open an issue.

> I would recommend this dataset:

Thanks @thewtex , I've confirmed I can read the 4D dataset in C++. It is a bit more difficult in ITK Python as `itk.Image` is wrapped only for 2D and 3D images by default. Attempting to read the 4D image from a 3D `itk.ImageFileReader` in Python currently throws an error:
```py
D:\repos\ITKNGFF-build\_deps\tensorstore-src\tensorstore/util/result.h:666: Status not ok: status(): INVALID_ARGUMENT: Number of dimensions (4) does not match number of indices (3)
```
It seems appropriate to fail to read the 4D image as 3D. In your and @dzenanz 's view, is this the correct reader behavior, or how should it be updated?

Reply:
In the Python package builds we do have support for 4D images (Image's and VectorImage's I believe). If you are testing locally with your own build, it is only 2D and 3D.

Reply:
Support for vector images is also on our TODO list: #32.

Reply:
> In the Python package builds we do have support for 4D images 

Ah, okay. Yes, since this is a development build I have currently wrapped only for 2D and 3D images.

> Support for vector images is also on our TODO list: https://github.com/InsightSoftwareConsortium/ITKIOOMEZarrNGFF/issues/32

Ah, thanks for pointing this out @dzenanz . @thewtex , maybe there is a 3D dataset that we could use for testing? Note that I have been successful in reading "5D" OME-Zarr data with size "1" along "t" and "c" dimensions.

Reply:
It looks like https://github.com/ome/ome-ngff-prototypes contains 2D and 3D single-channel images on S3 that can be read in under current tensorstore support. I will add testing for reading from these stores.

Reply:
Updated. Automated C++ testing now includes a test to fetch 2D and 3D data and validate with inline metadata.